src/components/Calculator.js

```diff
@@ -1,25 +1,37 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import NumberInput from './numberInput';
+import calculate from '../logic/calculate';
 
-const ButtonList = ({ items }) => (
-  <div className="calculatorButtons">
-    {items.map((item) => (
-      <button type="button" id={`button${item}`} key={item} className="button">{item}</button>
-    ))}
-  </div>
-);
+const Calculator = () => {
+  const [state, setState] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
 
-ButtonList.propTypes = {
-  items: PropTypes.arrayOf(PropTypes.string).isRequired,
-};
+  function handleClick(buttonName) {
+    const newState = calculate(state, buttonName.buttonText);
+    setState(newState);
+  }
+
+  const CalculatorButton = ({ buttonText }) => (
+    <button onClick={() => handleClick({ buttonText })} type="button" id={`button${buttonText}`} key={buttonText} className="button">{buttonText}</button>
+  );
+
+  CalculatorButton.propTypes = {
+    buttonText: PropTypes.string.isRequired,
+  };
```

Comment:
- [ ] It would be better to move this component to another file, this will make your code more readable, one of the goal of this milestone is to enable you learn how to pass props directly to any child component of the Calculator.js, great work so far 👍🏿